ts/packages/knowledgeProcessor/src/knowledgeIndex.ts

```diff
@@ -147,6 +154,10 @@ export type TextIndexSettings = {
     embeddingModel?: TextEmbeddingModel | undefined;
 };
 
+// There are *three* important types here:
+// - entries are always strings; typically words or sentences
+// - postings are arrays of TSourceIds; typically unique IDs for other objects
+// - TextId is a string that uniquely identifies an (entry, postings) pair internally
```

Comment:
Is this even right? I am still utterly confused about the meaning of TTextId and TSourceId in TextIndex. Is TTextId there a string that you can feed to an embedding (e.g. a topic or a query) or is it the type of ID generated for this index by the semantic index's put() method? Maybe the naming is just confusing?